Release v8.11.0 (#1187)
newrelic/node-newrelic: NEWS.md

### v8.11.0 (2022-05-11)

> * Log forwarding from APM agent is now possible.

Reviewer: true but that's not what this feature was completely.  I'd say perhaps `Application Logging in agent for winston now available`. Would like to get @coreyarnold feedback here

Reviewer: might want to call out it can do log forwarding, adding log metrics and local log decoration

    * Added application log aggregator to collect logs with adaptive sampling and appropriate max samples stored.

    * Added `application_logging` configuration and env vars with appropriate defaults.

>     * Changed default value of `application_logging.enabled` to be true.

Reviewer: this is a little weird here.  because we haven't released application_logging as a config yet so how could it change before release?

Reviewer 2: Agree, either strike this or change to "Set default value..."

>     * Set `application_logging.forwarding.enabled` to false when HSM is set.

Reviewer: i'm not sure people know what HSM is, maybe spell out high security mode

Reviewer 2: Since we might use HSM elsewhere, go with "High Security Mode (HSM)" on any first occurrence.

    * Enabled log forwarding by default in the example config.

    * Added sent, seen and dropped metrics that collected on every harvest cycle around log lines.

    * Added supportability metrics for some popular logging frameworks.
    
    * Added supportability metrics to record if the logging features are enabled.

    * Added a storage mechanisms to transactions to keep logs until transaction ends.

>     * Updated log aggregator to check transaction context before adding.

Reviewer: probably irrelevant, details of the churn of this feature

Reviewer 2: agreed. let's make sure the notes reflect the final product with sufficient detail.

>     * Ensured both `application_logging.enabled` and `application_logging.forwarding.enabled` are true before registering the log aggregator.

Reviewer: probably irrelevant to users

* Removed distributed tracing setting from example config

> * Fixed a typo in lib/instrumentation/core/child_process.js

Reviewer: this was a community PR we typically give shout outs to the PR author, you can see how we do that if you look down in the NEWS.md

Reviewer 2: https://github.com/newrelic/node-newrelic/pull/1171
Thanks @eitoball

> * Support automatic instrumentation of Redis v4

Reviewer: ```suggestion
* Added instrumentation support for `redis` v4.
```

* Bumped [moment](https://github.com/moment/moment) from 2.29.1 to 2.29.2.

* Bumped `tap` to 16.x.

> * Updated `ansi-regex` to resolve an audit warning from a CVE.

Reviewer: ```suggestion
* Updated `ansi-regex` to resolve a dev dependency audit warning.
```
